### PR TITLE
Finish user profiles

### PR DESCRIPTION
**Pull request by danielcrookall**
Users can search for and view other user profiles. 

- Homepage no longer exists (replaced with user page)
- User search now displays no results helper text.
- Clicking on Anagram button on top navigation takes user back to their profile (also added top navbar redirects).
- JWT now also stores username along with user id of logged in user.

### CODE REVIEW

**Comment by 0xla** [comment]
I was just testing the app, great work Daniel! I found a bug though. If the search bar is input with any random string, which is not present in the database, and the enter key is pressed, then it loads a user profile with that particular random string. 
Screen Shot 2022-07-26 at 1 49 50 AM

**Comment by danielcrookall** [comment]
Good point, I missed that! Will try and fix.

**Comment by 0xla** [comment]
> Good point, I missed that! Will try and fix.
> 
> Edit: also noticed users could manually edit URL and get taken to pages that shouldn't exist. Implemented a not found page as well.

Great ! It works perfectly now

**Review by 0xla — APPROVED** [review]

**Comment by swongg** [comment]
This is incredible 💯

**Review by swongg — APPROVED** [review]
💯

**Review by swongg — CHANGES_REQUESTED** [review]
When im logged in as Mango12345 and go on Daniel's page and upload an image from there, the image will be uploaded to Daniel's page. However, when I refresh the page, the newly uploaded image will be gone and if I navigate to mango12345's page, the uploaded image will be there. Seems like a redux issue with the states 🤔 (backend is fine imo)

**Comment by 0xla** [comment]
> When im logged in as Mango12345 and go on Daniel's page and upload an image from there, the image will be uploaded to Daniel's page. However, when I refresh the page, the newly uploaded image will be gone. If I navigate back to mango12345's page, the uploaded image will be there.
> 
> I think currently, whenever a new image is uploaded, it updates the images array redux state. I guess we need some type of check to determine whether the image url should be added to the state. (i.e. if the page being viewed is not the user's page, then we don't add a new image url to the array after uploading a new image). Alternatively, we could remove the camera upload button if the user is not viewing their own page. Similar to how we want to remove the follow button if the user is **on** their own page.
> 
> Maybe this could be added as a new github issue as a bug if it's not a quick fix 🤔

Good catch Steven this. I was just working on this actually. I have made it so that the follow button and the camera upload button will be removed when the user is not on their page.

**Comment by danielcrookall** [comment]
I think the best option here is just to remove the image upload if you're not on your page.

File: client/src/components/TopNavigation.tsx [hunk 1]
```diff
@@ -14,12 +15,22 @@ const TopNavigation = () => {
         navigate("/signIn")
     }
 
+    // @ts-ignore
+    const loggedInUsername = decodeToken(localStorage.getItem("authToken")).username;
```
[the remark this answers is not in the document]
Comment (danielcrookall) [reply client/src/components/TopNavigation.tsx:19]: You definitely could, I just don't know if it matters which one you use. What do you think?

**Review by swongg — APPROVED** [review]
LGTM

**Pull request merged**